splunk_connect_for_snmp_poller/manager/validator/inventory_validator.py

```diff
@@ -0,0 +1,104 @@
+import logging
+
+logger = logging.getLogger(__name__)
+
+INVENTORY_COMPONENTS_PER_LINE = 5
+SNMP_VERSION_1 = "1"
+SNMP_VERSION_2C = "2c"
+SNMP_VERSION_3 = "3"
+snmp_allowed_versions = {SNMP_VERSION_1, SNMP_VERSION_2C, SNMP_VERSION_3}
+
+
+def should_process_inventory_line(host_from_inventory):
+    stripped = host_from_inventory.lstrip()
+    return True if stripped and stripped[:1] != "#" else False
+
+
+def is_valid_number(port, validation):
+    try:
+        integer_value = int(port)
+        return validation(integer_value)
+    except ValueError:
+        logger.error(f"{port} is not a number")
+        return False
+
+
+def is_valid_port(port):
+    def up_to_65535(p):
+        return 1 <= p <= 65535
+
+    return is_valid_number(port, up_to_65535)
+
+
+def is_valid_second_quantity(seconds):
+    def any_positive_number(positive_number):
+        return positive_number > 0
+
+    return is_valid_number(seconds, any_positive_number)
+
+
+def resolve_host(hostname):
```

Comment:
Not sure if we should place host resolution in validation. We have to lookup DNS to confirm it and it can vary.
The question is whether we want to skip the host in case of no response from DNS or allow error messages in Splunk that the host in not resolvable later.

Reply:
Fixed. Moved to the test code